presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```
@@ -332,6 +333,16 @@ private void processNewSource(TaskSource source)
     {
         checkLockHeld("Lock must be held to call processInternal");
 
+        // revoke operators memory
+        // do this synchronously for now
+        for (int i = 0; i < operators.size() && !driverContext.isDone(); i++) {
+            Operator current = operators.get(i);
+            if (current.getOperatorContext().isSystemMemoryRevokingRequested()) {
+                getUnchecked(current.revokeMemory());
```

is there some way to do this without swallowing the errors that the Future might have?

---

It does not swallow. It will rethrow error as `RuntimeException` if there is some